app/views/unattended/provisioning_templates/provision/autoyast_default.erb

```diff
@@ -223,12 +243,72 @@ rm /etc/resolv.conf
         </source>
       </script>
     </chroot-scripts>
+<% if puppet_enabled -%>
+    <pre-scripts config:type="list">
+      <script>
+        <filename>puppet-keys.sh</filename>
+        <interpreter>shell</interpreter>
+        <notification>Retrieving Puppet repository keys ...</notification>
+        <source><![CDATA[
+<%
+  # Pull Puppetlabs RPM keys manually, automatic Zypper handling with key IDs does not work
+  puppet_repo_url_base = 'http://yum.puppet.com'
+
+  %w[puppet puppetlabs puppet-20250406].each { |key|
+-%>
+rpmkeys --import http://yum.puppet.com/RPM-GPG-KEY-<%= key %>
```

Comment:
This would only work, if the host has access to the internet. 

Do you have another idea how to solve this?

Reply:
Very good question. During analyzing this, I found the snippet https://github.com/theforeman/foreman/blob/84efe791cfe027992c213abb322773f11e136716/app/views/unattended/provisioning_templates/snippet/puppetlabs_repo.erb - Maybe this snippet can somehow extended so that it can be used for autoyast, too. Then there wouldn't be so much duplicate code.

Unfortunately, I don't see other sections in which the rpm keys for puppet are imported. Are they really necessary to install the puppet RPMs? Would it make sense to have them optional? Does `rpmkeys` work with proxy settings?

Reply:
I'll throw another look at it, but unfortunately SUSE installs can be a little more finicky in regards to installing packages with scripts - as AutoYaST holds the RPM lock during the entire install process.

It might be possible to attach the puppetlabs repo definition as an add-on repo set to allow all keys, install the release RPM as a regular system package - which will pull in the GPG keys, and then use the puppetlabs_repo snippet to remove the temporary "allow all keys" repository.